torchao/prototype/low_bit_optim/subclass_fp8.py

```diff
@@ -62,8 +74,13 @@ def dequantize(self, output_dtype=None):
         float_data = self.codes.float()
         float_data = float_data.view(-1, self.block_size) * self.scale.view(-1, 1)
 
+        if self.dynamic_range_expansion:
+            float_data = float_data.view(-1, self.block_size)
+            float_data =  float_data ** (1 / self.k.view(-1, 1))
```

Comment:
If we let `k=None` when `dynamic_range_expansion=False`, we can just check if `k is not None` here.